checks.d/docker_daemon.py

```diff
@@ -684,6 +697,24 @@ def _format_disk_metrics(self, metrics):
                     self.log.error('Unrecognized unit %s for disk metric %s. Dropping it.' % (unit, name))
         return metrics
 
+    def _calc_percent_disk_stats(self, stats):
+        """Calculate a percentage of used disk space for data and metadata"""
+        mtypes = ['data', 'metadata']
+        percs = {}
+        for mtype in mtypes:
+            used = stats.get('docker.{0}.used'.format(mtype))
+            total = stats.get('docker.{0}.total'.format(mtype))
+            free = stats.get('docker.{0}.free'.format(mtype))
+            if total < free + used:
+                self.log.error('used, free, and total disk metrics may be wrong, '
+                               'unable to calculate percentage.')
+                return {}
+            if used > 0.0 and total > 0.0:
```

Comment:
Might be worth looking at handling total = 0.0 simply as an exception. Not sure what might look cleaner.